modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/leases/Lease.java

```diff
@@ -95,7 +111,20 @@ public Lease acceptLease(HybridTimestamp to) {
                 + "leaseholder=" + leaseholder
                 + ", expirationTime=" + expirationTime + ']';
 
-        return new Lease(leaseholder, startTime, to, true);
+        return new Lease(leaseholder, startTime, to, true, true);
+    }
+
+    /**
+     * Denies the lease.
+     *
+     * @return Denied lease.
+     */
+    public Lease denyLease() {
+        assert accepted : "The lease is not accepted ["
+                + "leaseholder=" + leaseholder
+                + ", expirationTime=" + expirationTime + ']';
+
+        return new Lease(leaseholder, startTime, expirationTime, false, true);
```

Comment:
Accepted lease means we cannot change their expiration timestamp. Moreover, it is worth to refers to accepted flag like a storage marker (all leases with accepted flag are available for placement driver, but not available for other nodes). And here we shouldn't to remove an active lease (not expired) from one storage, because it should be available for all until their time is not elapsed.

Reply:
Well, I am wrong here about accepted flag - but still think that logging on lease denial after successful invoke might be useful